src/Facility.Core/ServiceJsonUtility.cs

```diff
@@ -11,13 +12,17 @@ namespace Facility.Core
 	/// <summary>
 	/// Helper methods for working with JSON.
 	/// </summary>
-	public static class ServiceJsonUtility
+	public static partial class ServiceJsonUtility
 	{
 		/// <summary>
 		/// Serializes a value to JSON.
 		/// </summary>
-		public static string ToJson(object? value)
+		public static string ToJson(object? value, FacilitySerializer serializer = default)
 		{
+#if NET6_0_OR_GREATER
+			if (serializer is FacilitySerializer.SystemTextJson || value is System.Text.Json.Nodes.JsonNode || value?.GetType().GetCustomAttribute<FacilitySerializerAttribute>() is { Value: FacilitySerializer.SystemTextJson })
```

Comment:
We should refactor the attribute check into a helper method. DRY, but also we might want a type cache if the attribute lookup is expensive.

Reply:
Yes, this is super messy as is. I'm not sure the attribute is the way to go anyway. I started with that, but it didn't work in some cases where serialization didn't start at a DTO, so I ended up with the `serializer` parameter.

Reply:
Yeah, you'd probably have to recurse through generic parameters, or at least check for known generic types.

Reply:
At this point I'm thinking about just checking the `serializer` and not the type.

Reply:
I haven't thought it all the way through, but should we obsolete `ServiceJsonUtility` and make `FacilitySerializer` (`FacilityJsonSerializer`?) an abstract class with `FromJson`, `ToJson`, etc.? I'd want to make it difficult to use the wrong serializer, i.e. I'd want to force the decision on callers. Even if we don't obsolete `ServiceJsonUtility`, I think I'd want to obsolete the overloads without a `FacilitySerializer`.

Reply:
Code generators wouldn't necessarily need to pass the serializer around, e.g. it could use `NewtonsoftFacilityJsonSerializer.Instance.FromJson`, and `NewtonsoftFacilityJsonSerializer` could have non-abstract Json.NET-specific methods on it. Same with `SystemFacilityJsonSerializer`.

Reply:
For consistency with other names in `Facility.Core`, `ServiceJsonSerializer` might be better than `FacilityJsonSerializer`.

Reply:
Would the abstract base class approach make it less likely that Newtonsoft.Json code would be JITted?